src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -177,15 +185,38 @@ private CloudBlockBlob acquireLease(CloudBlockBlob cloudBlockBlob, String contai
             }
             cloudBlockBlob.acquireLease(blobLeaseTimeout, null);
             return cloudBlockBlob;
-        });
+        } catch (StorageException storageException) {
+            if (storageException.getHttpStatusCode() == HttpStatus.CONFLICT.value()) {
+                log.error(
+                    "Lease already acquired for container {} and zip file {}",
+                    containerName,
+                    zipFilename,
+                    storageException
+                );
+            } else {
+                log.error(storageException.getMessage(), storageException);
+            }
+            return null;
+        } catch (Exception exception) {
+            log.error(
+                "Failed to acquire lease on file {} in container {}",
+                zipFilename,
+                containerName,
+                exception
+            );
+
+            return null;
+        }
     }
 
     private ZipFileProcessor processZipFileContent(
         ZipInputStream zis,
         String zipFilename,
         String containerName
     ) {
-        return errorWrapper.wrapDocFailure(containerName, zipFilename, () -> {
+        ZipFileProcessor processor = null;
+
+        try {
```

Comment:
I personally would pull out the bit in the try into a new method so this function is really just the error handling case.